lib/src/extensions/rect_extension.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:fl_chart/src/extensions/offset_extension.dart';
+import 'package:flutter/material.dart';
+
+extension RectExtension on Rect {
+  Rect applyPadding(EdgeInsets padding) {
+    return Rect.fromLTRB(
+      left - padding.left,
+      top - padding.top,
+      right + padding.right,
+      bottom + padding.bottom,
+    );
+  }
+
+  Rect applyMargin(EdgeInsets margin) => Rect.fromCenter(
```

Comment:
It seems this function doesn't work correctly.
Check the samples below:

---
```
tooltipMargin: const EdgeInsets.all(12), // Doesn't apply
```
<img width="391" alt="image" src="https://github.com/imaNNeo/fl_chart/assets/7009300/d34b05ce-92d2-4442-b137-3fd5e06bce1b">

---
```
tooltipMargin: const EdgeInsets.only(bottom: 12), /// Applies
```
<img width="386" alt="image" src="https://github.com/imaNNeo/fl_chart/assets/7009300/2a36029e-210f-4937-8c19-f510e84cd8ed">

Reply:
It seems `tooltipMargin` is useless when we already have `tooltipHorizontalOffse` and `tooltipVerticalOffset` properties. What do you think?